src/lv_widgets/lv_chart.c

```diff
@@ -1120,6 +1213,76 @@ static void draw_series_column(lv_obj_t * chart, const lv_area_t * series_area,
     }
 }
 
+/**
+ * Draw the cursors as lines on a chart
+ * @param obj pointer to chart object
+ */
+static void draw_cursors(lv_obj_t * chart, const lv_area_t * series_area, const lv_area_t * clip_area)
+{
+    lv_area_t series_mask;
+    bool mask_ret = _lv_area_intersect(&series_mask, series_area, clip_area);
+    if(mask_ret == false) return;
+
+    lv_chart_ext_t * ext = lv_obj_get_ext_attr(chart);
+    if(_lv_ll_is_empty(&ext->cursors_ll)) return;
+
+    lv_point_t p1;
+    lv_point_t p2;
+    lv_chart_cursor_t * cursor;
+
+    lv_draw_line_dsc_t line_dsc;
+    lv_draw_line_dsc_init(&line_dsc);
+    lv_obj_init_draw_line_dsc(chart, LV_CHART_PART_CURSOR, &line_dsc);
+
+    lv_draw_rect_dsc_t point_dsc;
+    lv_draw_rect_dsc_init(&point_dsc);
+    point_dsc.bg_opa = line_dsc.opa;
+    point_dsc.radius = LV_RADIUS_CIRCLE;
+
+    lv_coord_t point_radius = lv_obj_get_style_size(chart, LV_CHART_PART_CURSOR);
+
+    /*Do not bother with line ending is the point will over it*/
+    if(point_radius > line_dsc.width / 2) line_dsc.raw_end = 1;
+
+    /*Go through all cursor lines*/
+    _LV_LL_READ_BACK(ext->cursors_ll, cursor) {
+	line_dsc.color = cursor->color;
+	point_dsc.bg_color = cursor->color;
+
+	if(cursor->axis | LV_CHART_CURSOR_DIRECTION_HOR) {
+		p1.x = series_area->x1;
+		p1.y = series_area->y1 + cursor->point.y;
+		p2.x = p1.x + cursor->point.x;
```

Comment:
So the length of the cursor will be `cursor->point.x`?

Reply:
the cursor will be drawn as a solid/dashed line based on its direction from the axes. So when its direction is horizontal yeah it is its length. If it was vertical the point.y became its length. and if we or horizontal and vertical it will draw both of the lines, crossed at the point coordination.